src/echo/echo.rs

```diff
@@ -36,119 +34,82 @@ static HELP: &str = r#"
  \\xHH    byte with hexadecimal value HH (1 to 2 digits)
 "#;
 
-enum Base {
-    B8 = 8,
-    B16 = 16,
-}
-
-struct Opts {
-    newline: bool,
-    escape: bool,
+pub fn parse_code(input: &mut std::iter::Peekable<std::str::Chars<'_>>, base: u32, max_digits: u32, bits_per_digit: u32) -> char {
```

Comment:
This doesn't need to be `pub`.  Also, can you do `use std::iter::Peekable;` and `use std::str::Chars;` above instead of having the cruft in the function declaration?